modules/miner/miner.go

```diff
@@ -194,10 +199,15 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
```

Comment:
Yeah, can confirm that `miner.Close` only completes when `tg.IsStopped()` is checked within the loop. 

Should the test I wrote to confirm this be included in the PR?  See below.  It's slow as is, and 

```go
func TestMinerCloseDeadlock(t *testing.T) {
	if testing.Short() {
		t.SkipNow()
	}
	mt, err := createMinerTester("TestMinerCloseDeadlock")
	if err != nil {
		t.Fatal(err)
	}

	// Turn miner on.
	mt.miner.StartCPUMining()

        // Without this wait, Close() is called on the miner before StartCPUMining() 
        // manages to call 'go threadedMine()'.  If this function is going to be included
        // in the Sia test suite, I should find a real fix since this still sets off warnings 
        // about data races.
	time.Sleep(time.Second)

	// Call close without stopping the miner.  Since 'tg.IsStopped' is
        // with each iteration of the loop in 'threadedMine', the miner
        // should close on the next iteration of said loop.  Without this
        // check, Close would hang forever.
	if err = mt.miner.Close(); err != nil {
		t.Fatal(err)
	}
}
```